src/ucp/am/eager_single.c

```diff
@@ -72,9 +72,9 @@ ucp_am_eager_short_proto_progress_common(uct_pending_req_t *self, int is_reply)
     status = uct_ep_am_short_iov(ucp_ep_get_fast_lane(req->send.ep,
                                                       spriv->super.lane),
                                  am_id, iov, iov_cnt);
-    status = ucp_proto_am_handle_user_header_send_status(req, status);
     if (ucs_unlikely(status == UCS_ERR_NO_RESOURCE)) {
         req->send.lane = spriv->super.lane; /* for pending add */
+        status = ucp_am_handle_user_header_send_status_or_abort(req, status);
```

Comment:
1. maybe add new helper function "ucp_proto_am_req_copy_header_or_abort", and call it also from "ucp_am_handle_user_header_send_status_or_abort". to avoid extra check of the status.
2. no need to assign to 'status' - just return the function result directly

Reply:
Regarding option 1. I've tried it. I didn't liked the result because this function has hidden assumption. it can only be called for requests with `UCS_ERR_NO_RESOURCE` status.
I think we can count on dead code elimination here.